src/components/FilePicker.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="uk-height-1-1 uk-flex uk-flex-column uk-overflow-hidden">
+    <list-header
+      :current-folder="currentFolder"
+      :is-select-btn-enabled="isSelectBtnEnabled"
+      :is-location-picker="variation === 'location'"
+      @openFolder="loadFolder"
+      @select="emitSelectedResources"
+    />
+    <div
+      v-if="state === 'loading'"
+      key="loading-message"
+      class="uk-flex uk-flex-1 uk-flex-middle uk-flex-center"
+    >
+      <oc-spinner aria-label="Loading resources" />
+    </div>
+    <list-resources
+      v-if="state === 'loaded'"
+      key="resources-list"
+      class="uk-flex-1 oc-border"
+      :resources="resources"
+      :current-folder="currentFolder"
+      :is-location-picker="variation === 'location'"
+      @openFolder="loadFolder"
+      @selectResources="selectResources"
+    />
+  </div>
+</template>
+
+<script>
+import { buildResource } from '../helpers/resources'
+import ListResources from './ListResources.vue'
+import ListHeader from './ListHeader.vue'
+
+export default {
+  name: 'FilePicker',
+
+  components: {
+    ListHeader,
+    ListResources
+  },
+
+  props: {
+    variation: {
+      type: String,
+      required: true,
+      validator: value => value === 'resource' || 'location'
+    }
+  },
+
+  data: () => ({
+    state: 'loading',
+    resources: [],
+    currentFolder: null,
+    isSelectBtnEnabled: false,
+    selectedResources: [],
+    davProperties: [
+      '{http://owncloud.org/ns}permissions',
+      '{http://owncloud.org/ns}favorite',
+      '{http://owncloud.org/ns}fileid',
+      '{http://owncloud.org/ns}owner-id',
+      '{http://owncloud.org/ns}owner-display-name',
+      '{http://owncloud.org/ns}share-types',
+      '{http://owncloud.org/ns}privatelink',
+      '{DAV:}getcontentlength',
+      '{http://owncloud.org/ns}size',
+      '{DAV:}getlastmodified',
+      '{DAV:}getetag',
+      '{DAV:}resourcetype'
+    ]
+  }),
+
+  created() {
+    this.loadFolder('/')
```

Comment:
I guess later on we can have arguments to specify the default path to start with.
It will be useful whether for example someone is selecting multiple attachments for an email, they wouldn't want the picker to reset back to the root every time it's opened.

Reply:
Opened a ticket for this #8